### PR TITLE
Ganma: cleanup for chapter list and preferences

### PR DESCRIPTION
**Pull request by stevenyomi**
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by alessandrojean** [comment]
Curiosity: any particular reason to switch from data classes to normal classes in the DTO?

**Comment by stevenyomi** [comment]
> Curiosity: any particular reason to switch from data classes to normal classes in the DTO?

It's just that I wasn't using any data class features and using normal classes may reduce compiled code size. (stripping the auto-generated `toString`, `equals`, etc)

File: src/ja/ganma/src/eu/kanade/tachiyomi/extension/ja/ganma/Ganma.kt [hunk 1]
```diff
@@ -75,7 +75,7 @@ open class Ganma : HttpSource(), ConfigurableSource {
 
     override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
         client.newCall(realMangaDetailsRequest(manga)).asObservableSuccess()
-            .map { mangaDetailsParse(it).apply { initialized = true } }
+            .map { mangaDetailsParse(it) }
```
Comment (alessandrojean) [inline src/ja/ganma/src/eu/kanade/tachiyomi/extension/ja/ganma/Ganma.kt:78]: I think it's better to keep the `initialized = true` as the app uses it to some internal checks too.

Reference of the base `HttpSource` class: https://github.com/tachiyomiorg/tachiyomi/blob/master/app/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt#L172-L184

Comment (stevenyomi) [reply src/ja/ganma/src/eu/kanade/tachiyomi/extension/ja/ganma/Ganma.kt:78]: I moved it to `toSMangaDetails()` in `mangaDetailsParse()`.

**Pull request merged**